app/uk/gov/hmrc/teamsandrepositories/persitence/TeamsAndReposPersister.scala

```diff
@@ -126,4 +129,13 @@ class MongoTeamsAndRepositoriesPersister @Inject()(mongoConnector: MongoConnecto
           case modified => Some(modified)
         }
       }
+
+  private def withArchivedRepositoryFiltering(teamsAndRepos: List[TeamRepositories],
```

Comment:
Preferable to do apply the filter in the db query

Reply:
Hi @colin-lamed I'm not sure if that's possible. The structure of the items stored in Mongo is as a TeamsAndRepositories object, not as a Repository object. As it's a Repository that is archived, I think we need to return the TeamsAndRepositories object and then we can do the filtering. Happy to be corrected if I've misunderstood the existing DB structure.